sklearn/utils/estimator_checks.py

```diff
@@ -238,17 +247,30 @@ def check_estimator(Estimator):
     will be run if the Estimator class inherits from the corresponding mixin
     from sklearn.base.
 
+    This test can be applied to classes or instances.
+    Classes currently have some additional tests that related to construction,
+    while passing instances allows the testing of multiple options.
+
     Parameters
     ----------
     Estimator : class
-        Class to check. Estimator is a class object (not an instance).
+        Estimator to check. Estimator is a class object or instance.
 
```

Comment:
We need  way to also specify an optional name to `check_estimator`. When not using default parameters, we will want to default to `str(estimator)`. Indeed, what we want name to be (by default? in all cases where an instance is passed?) is actually the repr after default param values are hidden by #7618.